samples/nrf9160/modem_shell/src/link/link_shell.c

```diff
@@ -870,9 +904,97 @@ int link_shell(const struct shell *shell, size_t argc, char **argv)
 				goto show_usage;
 			}
 			break;
+		case LINK_SHELL_OPT_SEARCH_CFG: {
+			int loop_integer = 0;
+
+			ret = sscanf(
+				optarg,
+				"%d,%hd,%hd",
+				&loop_integer,
+				&search_cfg.return_to_pattern,
+				&search_cfg.band_optimization);
+
+			if (loop_integer != 0 && loop_integer != 1) {
+				mosh_error(
+					"1st parameter of --search_cfg option must be 0 or 1 "
+					"but %d given", loop_integer);
+				return -EINVAL;
+			}
```

Comment:
Maybe this should be moved after the `ret` checks?

Reply:
True.